presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import io.airlift.slice.XxHash64;
+
+import java.math.BigInteger;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class QueryInfoUtils
+{
+    private QueryInfoUtils() {}
+
+    public static String computeQueryHash(String query)
+    {
+        if (query.isEmpty()) {
+            return "";
+        }
+
+        byte[] queryBytes = query.getBytes(UTF_8);
+        long queryHash = XxHash64.hash(new BigInteger(queryBytes).longValue());
```

Comment:
XxHash64's hash function does not accept `String` types as input. We need to convert the bytes to a long. To do this, we use the `BigInteger` class which has a `longValue() method` commonly used to convert a byte array to a long value.